### PR TITLE
View document refactor

### PR DESCRIPTION
**Pull request by danielsjoo**
## Link to Issue
Previously, targetUser held the variable undefined, and the this.props.targetUser always held the constant string "view". With this refactor along with some upstream parent changes, this should fix the problem. 

Later on, the backend should be edited to not accept the undefined data holder for targetUser.

### CODE REVIEW

File: src/components/Documents/MyDocuments.tsx [hunk 1]
```diff
@@ -95,6 +95,10 @@ class MyDocuments extends Component<Props, State> {
     return files.length > maxNumFiles;
   }
 
+  resetDocumentId = () => {
+    this.setState({ currentDocumentId: undefined });
```
Comment (crchong1) [inline src/components/Documents/MyDocuments.tsx:99]: Don't you also have to reset the current document name as well

File: src/components/Documents/ViewDocument.tsx [hunk 2]
```diff
@@ -15,24 +15,15 @@ interface Props {
   documentDate: string;
   documentUploader: string;
   targetUser: string;
+  resetDocumentId: ()=>null;
```
Comment (crchong1) [inline src/components/Documents/ViewDocument.tsx:18]: why does this resolve as null? should it be void then?

**Review by crchong1 — CHANGES_REQUESTED** [review]
left some comments

File: src/components/Documents/ViewDocument.tsx (hunk 2 above)
Comment (danielsjoo) [reply src/components/Documents/ViewDocument.tsx:18]: You're right, I didn't realize there was a difference between null, void, and undefined.

**Review by crchong1 — APPROVED** [review]
looks great!

**Pull request merged**